e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-payment.spec.js

```diff
@@ -0,0 +1,92 @@
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+
+    const report = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION)
+      .withId(REPORT_ID)
+      .withBankId(BANK_ID)
+      .build();
+    const feeRecordOne = FeeRecordEntityMockBuilder.forReport(undefined)
+      .withId(FEE_RECORD_ID_ONE)
+      .withFacilityId('111111111')
+      .withExporter('Exporter 1')
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withFeesPaidToUkefForThePeriod(100)
+      .withFeesPaidToUkefForThePeriodCurrency('JPY')
+      .withPaymentExchangeRate(2)
+      .withStatus('TO_DO')
+      .build();
+    const feeRecordTwo = FeeRecordEntityMockBuilder.forReport(undefined)
+      .withId(FEE_RECORD_ID_TWO)
+      .withFacilityId('22222222')
+      .withExporter('Exporter 2')
+      .withFeesPaidToUkefForThePeriod(200)
+      .withFeesPaidToUkefForThePeriodCurrency('EUR')
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withPaymentExchangeRate(0.5)
+      .withStatus('TO_DO')
+      .build();
+    report.feeRecords = [feeRecordOne, feeRecordTwo];
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`utilisation-reports/${REPORT_ID}`);
+    cy.get(`[type="checkbox"][id="feeRecordId-${FEE_RECORD_ID_ONE}-reportedPaymentsCurrency-${PAYMENT_CURRENCY}-status-${FEE_RECORD_STATUS}"]`).check();
+    cy.get(`[type="checkbox"][id="feeRecordId-${FEE_RECORD_ID_TWO}-reportedPaymentsCurrency-${PAYMENT_CURRENCY}-status-${FEE_RECORD_STATUS}"]`).check();
+    cy.get('[type="submit"]').contains('Add a payment').click();
+  });
+
+  it('should render the add a payment page', () => {
+    cy.get('h1').invoke('text').should('contain', 'Add a payment');
+  });
+
+  it('should render the selected fee record details', () => {
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('11111111');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('Exporter 1');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('JPY 100');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('GBP 50');
+
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('22222222');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('Exporter 2');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('EUR 200');
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('GBP 400');
+
+    pages.utilisationReportAddPaymentPage.selectedReportedFeesDetailsTable().contains('Total reported payments GBP 450');
+  });
+
+  it('should display errors when form submitted with invalid values', () => {
+    cy.getInputByLabelText('Amount received').type('100');
```

Comment:
I recognise this is not the most traditional cypress test pattern, I generally prefer writing tests from the perspective of if a user where to visit the page how do they identify landmarks and interact over using lots of test ids, because then the tests are going to be more realistic and close to user interactions. But, if we would prefer to stick to the 'data-cy' selection pattern then I am (somewhat reluctantly) willing to change this to that as I do appreciate that is what [cypress suggest themselves](https://docs.cypress.io/guides/references/best-practices).

Reply:
Although I do find it interesting that they recommend 'data-cy' as the best method for element selection and then [recommend the testing library extension](https://docs.cypress.io/guides/references/best-practices#Cypress-and-Testing-Library) which is the anti-pattern using user first element selection...